packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -274,8 +270,16 @@ gmaps.Icon? _gmIconFromBitmapDescriptor(BitmapDescriptor bitmapDescriptor) {
     // Grab the bytes, and put them into a blob
     final List<int> bytes = iconConfig[1]! as List<int>;
     // Create a Blob from bytes, but let the browser figure out the encoding
-    final Blob blob = Blob(<dynamic>[bytes]);
-    icon = gmaps.Icon()..url = Url.createObjectUrlFromBlob(blob);
+    final Blob blob;
+
+    assert(
+      bytes is Uint8List,
+      'The bytes for a BitmapDescriptor icon must be a Uint8List',
+    );
+
+    blob = Blob(<JSUint8Array>[(bytes as Uint8List).toJS].toJS);
```

Comment:
Oof, this could use a package:web helper? @srujzs? +1 to [this](https://github.com/dart-lang/web/issues/180)?

Reply:
Added a TODO for later

Reply:
A (slower) alternative could be `[bytes].jsify() as JSArray<BlobPart>`, but I'm not sure how to make this much more concise while still respecting recursive conversions - food for though for myself.